sacc/tracers.py

```diff
@@ -786,3 +786,319 @@ def from_tables(cls, table_list):
+        Otherwise, they are equal if names and the r-range and centers of the
+        bins, are equal."""
+        if not isinstance(other, BinRadiusTracer):
+            return False
+        return (
+            self.name == other.name
+            and self.lower == other.lower
+            and self.center == other.center
+            and self.upper == other.upper
+        )
+
+    def __init__(self, name: str, lower: float, upper: float, center: float, **kwargs):
+        """
+        Create a tracer corresponding to a single radial bin.
+
+        :param name: The name of the tracer
+        :param lower: The lower bound of the radius bin
+        :param upper: The upper bound of the radius bin
+        :param center: The value to use if a single point-estimate is needed.
+
+        Note that :python:`center` need not be the midpoint between
+        :python:`lower` and :python:`upper`'.
+        """
+        super().__init__(name, **kwargs)
+        self.lower = lower
+        self.upper = upper
+        self.center = center
+
+    @classmethod
+    def to_tables(cls, instance_list):
+        """Convert a list of BinRadiusTracers to a single astropy table
+
+        This is used when saving data to a file.
+        One table is generated with the information for all the tracers.
+
+        :param instance_list: List of tracer instances
+        :return: List with a single astropy table
+        """
+
+        names = ["name", "quantity", "lower", "upper", "center"]
+
+        cols = [
+            [obj.name for obj in instance_list],
+            [obj.quantity for obj in instance_list],
+            [obj.lower for obj in instance_list],
+            [obj.upper for obj in instance_list],
+            [obj.center for obj in instance_list],
+        ]
+
+        table = Table(data=cols, names=names)
+        table.meta["SACCTYPE"] = "tracer"
+        table.meta["SACCCLSS"] = cls.tracer_type
+        table.meta["EXTNAME"] = f"tracer:{cls.tracer_type}"
+        return [table]
+
+    @classmethod
+    def from_tables(cls, table_list):
+        """Convert an astropy table into a dictionary of tracers
+
+        This is used when loading data from a file.
+        One tracer object is created for each "row" in each table.
+
+        :param table_list: List of astropy tables
+        :return: Dictionary of tracers
+        """
+        tracers = {}
+
+        for table in table_list:
+            for row in table:
+                name = row["name"]
+                quantity = row["quantity"]
+                lower = row["lower"]
+                upper = row["upper"]
+                center = row["center"]
+                tracers[name] = cls(
+                    name,
+                    quantity=quantity,
+                    lower=lower,
+                    upper=upper,
+                    center=center,
+                )
+        return tracers
+
+
+class ClusterSurveyTracer(BaseTracer, tracer_type="cluster_survey"):  # type: ignore
```

Comment:
I think the sky area is only needed for the covariance estimation, not e.g. the theory predictions - is that right?  So far we've been passing metadata like that which isn't needed for MCMC separately to TJPCov, rather than as a tracer.

If I'm wrong and this is needed for MCMC then the sky area tracer isn't specific to clusters, so could you rename it?

Reply:
We need the area for the prediction too, so I renamed the tracer to `SurveyTracer`,

Reply:
Thanks, that's great, just my ignorance!